### PR TITLE
ddt removal in 3 cli modules

### PR DESCRIPTION
**Pull request by sthirugn**

### CODE REVIEW

File: tests/foreman/cli/test_subnet.py [hunk 1]
```diff
@@ -308,25 +290,20 @@ def test_negative_update_2(self, options):
...
-        gen_string(str_type='numeric'),
-        gen_string(str_type='latin1'),
-        gen_string(str_type='utf8'),
-    )
-    def test_positive_delete_1(self, test_name):
+        for options in invalid_addr_pools():
+            with self.subTest(options):
+                opts = {u'id': subnet['id']}
+                # generate pool range from network address
+                for key, val in options.iteritems():
```
Comment (oshtaier) [inline tests/foreman/cli/test_subnet.py]: not sure whether that statement will work properly (but it is more lack of knowledge from my side)

**Comment by oshtaier** [comment]
ACK pending test results to make sure that nothing is broken

File: tests/foreman/cli/test_subnet.py (hunk 1 above)
Comment (sthirugn) [inline tests/foreman/cli/test_subnet.py]: Yep this will work as the data generator function returns a tuple of dictionaries.

**Comment by sthirugn** [comment]
@oshtaier good point, my bad - did not paste the test results on friday.  I dont have the logs anymore, rerunning now and will paste in few minutes. Thank you for pointing this out.

**Comment by sthirugn** [comment]
Test results:
test_subnet.py

```py
Finding files... done.
Importing test modules ... done.

----------------------------------------------------------------------
Ran 15 tests in 169.099s

OK (skipped=1)

```

test_roles.py
```py
Finding files... done.
Importing test modules ... done.

----------------------------------------------------------------------
Ran 4 tests in 1.286s

OK (skipped=4)
```

test_subscription.py
```py
Finding files... done.
Importing test modules ... done.

----------------------------------------------------------------------
Ran 6 tests in 350.076s

OK (skipped=1)

```

**Comment by sthirugn** [comment]
Rebased

**Comment by oshtaier** [comment]
ACK

**Comment by sthirugn** [comment]
rebased again

**Comment by elyezer** [comment]
ACK

**Comment by sthirugn** [comment]
Part of https://github.com/SatelliteQE/robottelo/issues/2769

**Pull request merged**